Description: Spin the event loop before destroying the menu.
 This allows the menu implementation to complete toggling the
 launcher before the instance is destroyed. It's extremely un-
 fortunate that the library currently requires this careful
 handling; see e944d7ae in kde-workspace for a note on what
 needs to be cleaned up there.
 Author: Eike Hein <hein@kde.org>
Origin: upstream, https://projects.kde.org/projects/kde/kdeplasma-addons/repository/revisions/7c3011e0a6459c00cf9d06d74da3a32f87f1a2f6
Bug: https://bugs.kde.org/show_bug.cgi?id=324661
Applied-Upstream: 4.11.2
Reviewed-by: Howard Chan <smartboyhw@gmail.com>
Last-Update: 2013-09-15
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
diff --git a/applets/icontasks/windowtaskitem.cpp b/applets/icontasks/windowtaskitem.cpp
index 970c19f..61b39ef 100644
--- a/applets/icontasks/windowtaskitem.cpp
+++ b/applets/icontasks/windowtaskitem.cpp
@@ -406,17 +406,18 @@ void WindowTaskItem::showContextMenu(const QPoint &pos, bool showAppMenu)
         actionList.append(configAction);
     }
 
-    TaskManager::BasicMenu menu(0, m_task.data(), &m_applet->groupManager(), actionList, showAppMenu ? getAppMenu() : QList <QAction*>());
-    menu.adjustSize();
+    TaskManager::BasicMenu * menu = new TaskManager::BasicMenu(0, m_task.data(), &m_applet->groupManager(), actionList, showAppMenu ? getAppMenu() : QList <QAction*>());
+    menu->adjustSize();
 
     if (m_applet->formFactor() != Plasma::Vertical) {
-        menu.setMinimumWidth(size().width());
+        menu->setMinimumWidth(size().width());
     }
 
     Q_ASSERT(m_applet->containment());
     Q_ASSERT(m_applet->containment()->corona());
     stopWindowHoverEffect();
-    menu.exec(pos.isNull() ? m_applet->containment()->corona()->popupPosition(this, menu.size()) : pos);
+    menu->exec(pos.isNull() ? m_applet->containment()->corona()->popupPosition(this, menu->size()) : pos);
+    menu->deleteLater();
     delete a;
 }
 
